src/plugins/console/public/application/containers/editor/monaco/utils/status_code_decoration_utils.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { monaco } from '@kbn/monaco';
+import { RequestResult } from '../../../../hooks/use_send_current_request/send_request';
+import {
+  DEFAULT_STATUS_BADGE_CLASSNAME,
+  SUCCESS_STATUS_BADGE_CLASSNAME,
+  PRIMARY_STATUS_BADGE_CLASSNAME,
+  WARNING_STATUS_BADGE_CLASSNAME,
+  DANGER_STATUS_BADGE_CLASSNAME,
+} from './constants';
+
+const getStatusCodeClassName = (statusCode: number) => {
+  if (statusCode <= 199) {
+    return DEFAULT_STATUS_BADGE_CLASSNAME;
+  }
+  if (statusCode <= 299) {
+    return SUCCESS_STATUS_BADGE_CLASSNAME;
+  }
+  if (statusCode <= 399) {
+    return PRIMARY_STATUS_BADGE_CLASSNAME;
+  }
+  if (statusCode <= 499) {
+    return WARNING_STATUS_BADGE_CLASSNAME;
+  }
+  return DANGER_STATUS_BADGE_CLASSNAME;
+};
+
+export const getStatusCodeDecorations = (data: RequestResult[]) => {
+  const decorations: monaco.editor.IModelDeltaDecoration[] = [];
+  let lastRequestEndLine = 0;
```

Comment:
nit: I think that should be named `lastResponseEndLine`

Reply:
Nice catch! Fixed in [0538534](https://github.com/elastic/kibana/pull/189210/commits/0538534133858e47d7ef32a7674cf42a4d648ec7).